### ISSUE TITLE
💅  `noArrayIndexKey` is not detected in Fragment shorthand syntax

### ISSUE THREAD

**Issue by mecaota** [biomejs/biome#7700]
### Environment information

```bash
CLI:
  Version:                      2.2.4
  Color support:                true

Platform:
  CPU Architecture:             aarch64
  OS:                           macos

Environment:
  BIOME_LOG_PATH:               unset
  BIOME_LOG_PREFIX_NAME:        unset
  BIOME_CONFIG_PATH:            unset
  BIOME_THREADS:                unset
  NO_COLOR:                     unset
  TERM:                         xterm-256color
  JS_RUNTIME_VERSION:           v20.19.5
  JS_RUNTIME_NAME:              node
  NODE_PACKAGE_MANAGER:         yarn/4.9.4

Biome Configuration:
  Status:                       Loaded successfully
  Path:                         biome.jsonc
  Formatter enabled:            true
  Linter enabled:               true
  Assist enabled:               true
  VCS enabled:                  true

Linter:
  JavaScript enabled:           unset
  JSON enabled:                 unset
  CSS enabled:                  unset
  GraphQL enabled:              unset
  Recommended:                  true
  Enabled rules:
    a11y/noAccessKey
    a11y/noAriaHiddenOnFocusable
    a11y/noAriaUnsupportedElements
    a11y/noAutofocus
    a11y/noDistractingElements
    a11y/noHeaderScope
    a11y/noInteractiveElementToNoninteractiveRole
    a11y/noLabelWithoutControl
    a11y/noNoninteractiveElementInteractions
    a11y/noNoninteractiveElementToInteractiveRole
    a11y/noNoninteractiveTabindex
    a11y/noPositiveTabindex
    a11y/noRedundantAlt
    a11y/noRedundantRoles
    a11y/noStaticElementInteractions
    a11y/noSvgWithoutTitle
    a11y/useAltText
    a11y/useAnchorContent
    a11y/useAriaActivedescendantWithTabindex
    a11y/useAriaPropsForRole
    a11y/useAriaPropsSupportedByRole
    a11y/useButtonType
    a11y/useFocusableInteractive
    a11y/useGenericFontNames
    a11y/useHeadingContent
    a11y/useHtmlLang
    a11y/useIframeTitle
    a11y/useKeyWithClickEvents
    a11y/useKeyWithMouseEvents
    a11y/useSemanticElements
    a11y/useValidAnchor
    a11y/useValidAriaProps
    a11y/useValidAriaRole
    a11y/useValidAriaValues
    a11y/useValidAutocomplete
    a11y/useValidLang
    complexity/noAdjacentSpacesInRegex
    complexity/noArguments
    complexity/noBannedTypes
    complexity/noCommaOperator
    complexity/noEmptyTypeParameters
    complexity/noExcessiveCognitiveComplexity
    complexity/noExcessiveNestedTestSuites
    complexity/noExtraBooleanCast
    complexity/noFlatMapIdentity
    complexity/noImportantStyles
    complexity/noStaticOnlyClass
    complexity/noThisInStatic
    complexity/noUselessCatch
    complexity/noUselessConstructor
    complexity/noUselessContinue
    complexity/noUselessEmptyExport
    complexity/noUselessEscapeInRegex
    complexity/noUselessFragments
    complexity/noUselessLabel
    complexity/noUselessLoneBlockStatements
    complexity/noUselessRename
    complexity/noUselessStringConcat
    complexity/noUselessStringRaw
    complexity/noUselessSwitchCase
    complexity/noUselessTernary
    complexity/noUselessThisAlias
    complexity/noUselessTypeConstraint
    complexity/noUselessUndefinedInitialization
    complexity/noVoid
    complexity/useArrowFunction
    complexity/useDateNow
    complexity/useFlatMap
    complexity/useIndexOf
    complexity/useLiteralKeys
    complexity/useNumericLiterals
    complexity/useOptionalChain
    complexity/useRegexLiterals
    complexity/useSimpleNumberKeys
    complexity/useSimplifiedLogicExpression
    complexity/useWhile
    correctness/noChildrenProp
    correctness/noConstAssign
    correctness/noConstantCondition
    correctness/noConstantMathMinMaxClamp
    correctness/noConstructorReturn
    correctness/noEmptyCharacterClassInRegex
    correctness/noEmptyPattern
    correctness/noGlobalDirnameFilename
    correctness/noGlobalObjectCalls
    correctness/noInnerDeclarations
    correctness/noInvalidBuiltinInstantiation
    correctness/noInvalidConstructorSuper
    correctness/noInvalidDirectionInLinearGradient
    correctness/noInvalidGridAreas
    correctness/noInvalidPositionAtImportRule
    correctness/noInvalidUseBeforeDeclaration
    correctness/noMissingVarFunction
    correctness/noNestedComponentDefinitions
    correctness/noNonoctalDecimalEscape
    correctness/noPrecisionLoss
    correctness/noReactPropAssignments
    correctness/noRenderReturnValue
    correctness/noSelfAssign
    correctness/noSetterReturn
    correctness/noStringCaseMismatch
    correctness/noSwitchDeclarations
    correctness/noUndeclaredVariables
    correctness/noUnknownFunction
    correctness/noUnknownMediaFeatureName
    correctness/noUnknownProperty
    correctness/noUnknownPseudoClass
    correctness/noUnknownPseudoElement
    correctness/noUnknownTypeSelector
    correctness/noUnknownUnit
    correctness/noUnmatchableAnbSelector
    correctness/noUnreachable
    correctness/noUnreachableSuper
    correctness/noUnsafeFinally
    correctness/noUnsafeOptionalChaining
    correctness/noUnusedFunctionParameters
    correctness/noUnusedImports
    correctness/noUnusedLabels
    correctness/noUnusedPrivateClassMembers
    correctness/noUnusedVariables
    correctness/noVoidElementsWithChildren
    correctness/noVoidTypeReturn
    correctness/useGraphqlNamedOperations
    correctness/useIsNan
    correctness/useParseIntRadix
    correctness/useSingleJsDocAsterisk
    correctness/useValidForDirection
    correctness/useValidTypeof
    correctness/useYield
    performance/noAccumulatingSpread
    performance/noAwaitInLoops
    performance/noBarrelFile
    performance/noDelete
    performance/noDynamicNamespaceImportAccess
    performance/noReExportAll
    security/noBlankTarget
    security/noDangerouslySetInnerHtml
    security/noDangerouslySetInnerHtmlWithChildren
    security/noGlobalEval
    style/noCommonJs
    style/noDescendingSpecificity
    style/noDoneCallback
    style/noEnum
    style/noExportedImports
    style/noNamespace
    style/noNegationElse
    style/noNestedTernary
    style/noNonNullAssertion
    style/noParameterAssign
    style/noParameterProperties
    style/noSubstr
    style/noUnusedTemplateLiteral
    style/noUselessElse
    style/noValueAtRule
    style/useArrayLiterals
    style/useAsConstAssertion
    style/useBlockStatements
    style/useCollapsedElseIf
    style/useCollapsedIf
    style/useConsistentArrayType
    style/useConsistentBuiltinInstantiation
    style/useConsistentCurlyBraces
    style/useConsistentObjectDefinitions
    style/useConst
    style/useDefaultParameterLast
    style/useDefaultSwitchClause
    style/useDeprecatedReason
    style/useEnumInitializers
    style/useExponentiationOperator
    style/useExportType
    style/useForOf
    style/useFragmentSyntax
    style/useGroupedAccessorPairs
    style/useImportType
    style/useLiteralEnumMembers
    style/useNodeAssertStrict
    style/useNodejsImportProtocol
    style/useObjectSpread
    style/useReadonlyClassProperties
    style/useSelfClosingElements
    style/useShorthandAssign
    style/useShorthandFunctionType
    style/useSingleVarDeclarator
    style/useSymbolDescription
    style/useTemplate
    style/useThrowNewError
    style/useThrowOnlyError
    style/useTrimStartEnd
    style/useUnifiedTypeSignatures
    suspicious/noApproximativeNumericConstant
    suspicious/noArrayIndexKey
    suspicious/noAssignInExpressions
    suspicious/noAsyncPromiseExecutor
    suspicious/noBiomeFirstException
    suspicious/noBitwiseOperators
    suspicious/noCatchAssign
    suspicious/noClassAssign
    suspicious/noCommentText
    suspicious/noCompareNegZero
    suspicious/noConfusingLabels
    suspicious/noConfusingVoidType
    suspicious/noConstEnum
    suspicious/noConstantBinaryExpressions
    suspicious/noControlCharactersInRegex
    suspicious/noDebugger
    suspicious/noDoubleEquals
    suspicious/noDuplicateAtImportRules
    suspicious/noDuplicateCase
    suspicious/noDuplicateClassMembers
    suspicious/noDuplicateCustomProperties
    suspicious/noDuplicateElseIf
    suspicious/noDuplicateFields
    suspicious/noDuplicateFontNames
    suspicious/noDuplicateJsxProps
    suspicious/noDuplicateObjectKeys
    suspicious/noDuplicateParameters
    suspicious/noDuplicateProperties
    suspicious/noDuplicateSelectorsKeyframeBlock
    suspicious/noEmptyBlock
    suspicious/noEmptyInterface
    suspicious/noEvolvingTypes
    suspicious/noExplicitAny
    suspicious/noExtraNonNullAssertion
    suspicious/noFallthroughSwitchClause
    suspicious/noFunctionAssign
    suspicious/noGlobalAssign
    suspicious/noGlobalIsFinite
    suspicious/noGlobalIsNan
    suspicious/noImplicitAnyLet
    suspicious/noImportAssign
    suspicious/noImportantInKeyframe
    suspicious/noIrregularWhitespace
    suspicious/noLabelVar
    suspicious/noMisleadingCharacterClass
    suspicious/noMisleadingInstantiator
    suspicious/noMisplacedAssertion
    suspicious/noMisrefactoredShorthandAssign
    suspicious/noOctalEscape
    suspicious/noPrototypeBuiltins
    suspicious/noQuickfixBiome
    suspicious/noRedeclare
    suspicious/noRedundantUseStrict
    suspicious/noSelfCompare
    suspicious/noShadowRestrictedNames
    suspicious/noShorthandPropertyOverrides
    suspicious/noSkippedTests
    suspicious/noSparseArray
    suspicious/noSuspiciousSemicolonInJsx
    suspicious/noTemplateCurlyInString
    suspicious/noThenProperty
    suspicious/noTsIgnore
    suspicious/noUnassignedVariables
    suspicious/noUnknownAtRules
    suspicious/noUnsafeDeclarationMerging
    suspicious/noUnsafeNegation
    suspicious/noUselessEscapeInString
    suspicious/noUselessRegexBackrefs
    suspicious/noVar
    suspicious/noWith
    suspicious/useAdjacentOverloadSignatures
    suspicious/useBiomeIgnoreFolder
    suspicious/useDefaultSwitchClauseLast
    suspicious/useErrorMessage
    suspicious/useGetterReturn
    suspicious/useGoogleFontDisplay
    suspicious/useGuardForIn
    suspicious/useIsArray
    suspicious/useIterableCallbackReturn
    suspicious/useNamespaceKeyword
    suspicious/useNumberToFixedDigitsArgument
    suspicious/useStrictMode

Workspace:
  Open Documents:               0
```

### Rule name

noArrayIndexKey

### Playground link

https://biomejs.dev/playground/?bracketSameLine=true&indentScriptAndStyle=true&code=ZQB4AHAAbwByAHQAIABjAG8AbgBzAHQAIABOAG8ARABlAHQAZQBjAHQAIAA9ACAAKAApACAAPQA%2BACAAewAKACAAIABjAG8AbgBzAHQAIABhAHIAcgBhAHkAcwAgAD0AIABbACIAYQAiACwAIAAiAGIAIgAsACAAIgBjACIAXQA7AAoAIAAgAHIAZQB0AHUAcgBuACAAYQByAHIAYQB5AHMALgBtAGEAcAAoACgAdgApAD0APgAgACgACgAgACAAIAAgADwAPgB7AC8AKgAgAG4AbwAgAGQAZQB0AGUAYwB0ACAAKgAvAH0ACgAgACAAIAAgACAAIAA8AGgAMgA%2BAFQAaABpAHMAIABpAHMALgAuAC4APAAvAGgAMgA%2BAAoAIAAgACAAIAAgACAAPABwAD4AewB2AH0APAAvAHAAPgAKACAAIAAgACAAPAAvAD4ACgAgACAAKQApAAoAfQAKAAoAZQB4AHAAbwByAHQAIABjAG8AbgBzAHQAIABDAGEAbgBEAGUAdABlAGMAdAAgAD0AIAAoACkAIAA9AD4AIAB7AAoAIAAgAGMAbwBuAHMAdAAgAGEAcgByAGEAeQBzACAAPQAgAFsAIgBhACIALAAgACIAYgAiACwAIAAiAGMAIgBdADsACgAgACAAcgBlAHQAdQByAG4AIABhAHIAcgBhAHkAcwAuAG0AYQBwACgAKAB2ACkAPQA%2BACAAKAAKACAAIAAgACAAPABGAHIAYQBnAG0AZQBuAHQAPgB7AC8AKgAgAGQAZQB0AGUAYwB0ACAAKgAvAH0ACgAgACAAIAAgACAAIAA8AGgAMgA%2BAFQAaABpAHMAIABpAHMALgAuAC4APAAvAGgAMgA%2BAAoAIAAgACAAIAAgACAAPABwAD4AewB2AH0APAAvAHAAPgAKACAAIAAgACAAPAAvAEYAcgBhAGcAbQBlAG4AdAA%2BAAoAIAAgACkAKQAKAH0A&ruleDomains.react=all

### Expected result

Because we have `useFragmentSyntax` enabled, we use shorthand fragment syntax and rarely use Fragments. However, we have come across cases where we need to use a key. With Fragments written in shorthand syntax, even if you forget to write a key, an error due to `noArrayIndexKey` is not detected.
We were unable to notice this in CI, and it was detected during team review.

I would like errors due to `noArrayIndexKey` to be detected even in Fragments written in shorthand syntax.

### Code of Conduct

- [x] I agree to follow Biome's Code of Conduct

**Comment by siketyan** [comment]
The `noArrayIndexKey` rule doesn't report missing `key` props. These cases are covered by the `useJsxKeyInIterable` rule.

**Comment by siketyan** [comment]
I realised the `useJsxKeyInIterable` doesn't report an error in this case.

**Comment by siketyan** [comment]
eslint-plugin-react doesn't check shorthand fragments too, but it has an option that enables the check.

https://github.com/jsx-eslint/eslint-plugin-react/blob/master/docs/rules/jsx-key.md#checkfragmentshorthand-default-false

**Comment by siketyan** [comment]
We also have that option!

https://biomejs.dev/linter/rules/use-jsx-key-in-iterable/#checkshorthandfragments

Playground: https://biomejs.dev/playground/?bracketSameLine=true&indentScriptAndStyle=true&files.biome.json=ewAKACAAIAAiAGwAaQBuAHQAZQByACIAOgAgAHsACgAgACAAIAAgACIAZQBuAGEAYgBsAGUAZAAiADoAIAB0AHIAdQBlACwACgAgACAAIAAgACIAcgB1AGwAZQBzACIAOgAgAHsACgAgACAAIAAgACAAIAAiAHIAZQBjAG8AbQBtAGUAbgBkAGUAZAAiADoAIAB0AHIAdQBlACwACgAgACAAIAAgACAAIAAiAGMAbwByAHIAZQBjAHQAbgBlAHMAcwAiADoAIAB7AAoAIAAgACAAIAAgACAAIAAgACIAdQBzAGUASgBzAHgASwBlAHkASQBuAEkAdABlAHIAYQBiAGwAZQAiADoAIAB7AAoAIAAgACAAIAAgACAAIAAgACAAIAAiAGwAZQB2AGUAbAAiADoAIAAiAG8AbgAiACwACgAgACAAIAAgACAAIAAgACAAIAAgACIAbwBwAHQAaQBvAG4AcwAiADoAIAB7AAoAIAAgACAAIAAgACAAIAAgACAAIAAgACAAIgBjAGgAZQBjAGsAUwBoAG8AcgB0AGgAYQBuAGQARgByAGEAZwBtAGUAbgB0AHMAIgA6ACAAdAByAHUAZQAKACAAIAAgACAAIAAgACAAIAAgACAAfQAKACAAIAAgACAAIAAgACAAIAB9AAoAIAAgACAAIAAgACAAfQAKACAAIAAgACAAfQAKACAAIAB9AAoAfQAKAA%3D%3D&files.main.tsx=ZQB4AHAAbwByAHQAIABjAG8AbgBzAHQAIABOAG8ARABlAHQAZQBjAHQAIAA9ACAAKAApACAAPQA%2BACAAewAKACAAIABjAG8AbgBzAHQAIABhAHIAcgBhAHkAcwAgAD0AIABbACIAYQAiACwAIAAiAGIAIgAsACAAIgBjACIAXQA7AAoAIAAgAHIAZQB0AHUAcgBuACAAYQByAHIAYQB5AHMALgBtAGEAcAAoACgAdgApAD0APgAgACgACgAgACAAIAAgADwAPgB7AC8AKgAgAG4AbwAgAGQAZQB0AGUAYwB0ACAAKgAvAH0ACgAgACAAIAAgACAAIAA8AGgAMgA%2BAFQAaABpAHMAIABpAHMALgAuAC4APAAvAGgAMgA%2BAAoAIAAgACAAIAAgACAAPABwAD4AewB2AH0APAAvAHAAPgAKACAAIAAgACAAPAAvAD4ACgAgACAAKQApAAoAfQAKAAoAZQB4AHAAbwByAHQAIABjAG8AbgBzAHQAIABDAGEAbgBEAGUAdABlAGMAdAAgAD0AIAAoACkAIAA9AD4AIAB7AAoAIAAgAGMAbwBuAHMAdAAgAGEAcgByAGEAeQBzACAAPQAgAFsAIgBhACIALAAgACIAYgAiACwAIAAiAGMAIgBdADsACgAgACAAcgBlAHQAdQByAG4AIABhAHIAcgBhAHkAcwAuAG0AYQBwACgAKAB2ACkAPQA%2BACAAKAAKACAAIAAgACAAPABGAHIAYQBnAG0AZQBuAHQAPgB7AC8AKgAgAGQAZQB0AGUAYwB0ACAAKgAvAH0ACgAgACAAIAAgACAAIAA8AGgAMgA%2BAFQAaABpAHMAIABpAHMALgAuAC4APAAvAGgAMgA%2BAAoAIAAgACAAIAAgACAAPABwAD4AewB2AH0APAAvAHAAPgAKACAAIAAgACAAPAAvAEYAcgBhAGcAbQBlAG4AdAA%2BAAoAIAAgACkAKQAKAH0A&ruleDomains.react=all

**Comment by mecaota** [comment]
Thank you. I had the rule wrong. I set it up as you suggested and it worked fine.

However, although this rule is a recommended rule in React Domains, the default setting for checkShorthandFragments was false, so the omission went unnoticed. The default setting for a similar setting in eslint is false, but I don't think there's any problem with setting checkShorthandFragments to true by default in biomejs. This is because forgetting this rule will still result in a client error.
